### DISCUSSION TITLE
Would it be possible to show shortest unique prefix (omit common prefix)?

### DISCUSSION THREAD

**Discussion by piotr-dobrogost** [vlascik/Tabdir#6] [General]
Having the following two files opened in the editor (PyCharm):
`bgk/frontend/be_api_gtw/modules/products/bgk_be_modules/products/navigation.py`
`bgk/frontend/be_api_gtw/modules/products/bgk_be_modules/deposit_sums/navigation.py`

with `Max number of dirs to show: 3` in the plugin's settings
results in the following names in tabs:
`[modules|products|products] navigation.py`
`[modules|products|deposit_sums] navigation.py`

Why is the folder name `bgk_be_modules` being omitted?

Can the option be added to omit common prefix and present only the shortest unique prefix so that tabs would be named:
`[products] navigation.py`
`[deposit_sums] navigation.py`
as it's enough to know which file is which?

**Comment by vlascik** [comment]
I honestly have no idea how the diffing algorithm works here and what the plugin's original author meant by it, I just didn't want to remove it without reason. 

You can find it here though: https://github.com/vlascik/Tabdir/blob/47d65aab80a86070f927d8f85e18abaf4a77e29a/src/ru/crazycoder/plugins/tabdir/SameFilenameTitleProvider.java , maybe you can do some spelunking in a debugger.

Or you can use a combination of folder specific rules and regex chains to get at least closer to where you want the title to be.
